### DISCUSSION TITLE
Remember last selected filter values for a datasource and use them across dashboards

### DISCUSSION THREAD

**Discussion by durchgedreht** [apache/superset#18429] [Ideas]
When switching dashboards the Filterbox is loaded from scratch and the user sees the preselection of the Time picker, e.g. "Last week":

<img width="246" alt="Screenshot 2021-01-15 at 10 24 01" src="https://user-images.githubusercontent.com/4490840/104711442-ac3abd00-5721-11eb-8d25-771a53e4be26.png">

We have multiple dashboards and users switch between them and they also need to analyse day, week and month. So they always have to chnage daterange over and over again. It would be great if one could configure the dashboards to take over the selection from last dashboard if available (like if it was handed over as URL parameter).

If that filter is available, apply it, if not use behaviour as is. Potentially this could be store in a cookie or session. 

The filterbox component could be marked with a checkbox 'use last filter' or similar.

**Comment by ktmud** [comment]
You can actually persist default filters for dashboards, including the time filters.  It's just the option is a little hidden:

<img width="917" alt="dashboard-properties" src="https://user-images.githubusercontent.com/335541/108139378-d4953e80-7074-11eb-9a1a-fa8f05f5cd21.png">

You need to go to the edit mode,  Edit Dashboard Properties, and add a `default_filters` field of stringified JSON in the advanced settings.

**Comment by ktmud** [comment]
<img width="394" alt="share-dashboard" src="https://user-images.githubusercontent.com/335541/108139697-7ae14400-7075-11eb-8caa-bfdf2e9588f6.png">


<img width="538" alt="share-chart-with-filters" src="https://user-images.githubusercontent.com/335541/108139642-5b4a1b80-7075-11eb-9f2e-34dfe9d04925.png">

You can also apply the filter values you like, then click on "Share dashboard" or "Share chart" to get permlinks of the dashboard with filter values remembered.

**Comment by durchgedreht** [comment]
This is not what I've asked for:
This way you need to inject the filter EVERY time.

It would be great if you could keep your selection and walk along. Think about the user:

- Open Dashboard "sold items"
- Selecting "last week" in filters
- **WOW ! So many sales!!! I'd like to see my revenue!!!**
- Next he switches to dashboard "revenue"
- likely he's interested in the same timeframe

It would be cool to have an option "use last selection if available". That would override the default.

How can I persist the selection? How can Superset itself inject that the way you described?

**Comment by ktmud** [comment]
If multiple dashboards are related, then maybe you can place them in different tabs of the same dashboard, so the time filter can be reused. But I do admit there may be value in remembering last selected filters and use them across dashboards.
